### PR TITLE
Kousa/isaac/chat to pubsub

### PR DESCRIPTION
**Pull request by dark-elixir-shibe**

### CODE REVIEW

File: kousa/lib/broth/message/auth/request.ex [hunk 1]
```diff
@@ -134,6 +135,9 @@ defmodule Broth.Message.Auth.Request do
           :ok
       end
 
+      # subscribe to chats directed to oneself.
+      PubSub.subscribe("chat:" <> user_id)
```
Comment (ityonemo) [inline kousa/lib/broth/message/auth/request.ex:143]: when you're authenticated, you subscribe to any chat messages that are directed straight to you.

File: kousa/lib/kousa/room.ex [hunk 2]
```diff
@@ -418,6 +419,9 @@ defmodule Kousa.Room do
           end)
         end
 
+        # subscribe to this room's chat
+        Onion.PubSub.subscribe("chat:" <> id)
+
```
Comment (ityonemo) [inline kousa/lib/kousa/room.ex:424]: here you subscribe to the room's chat only.

File: kousa/lib/kousa/room.ex [hunk 3]
```diff
@@ -519,6 +526,9 @@ defmodule Kousa.Room do
           Onion.RoomSession.leave_room(current_room_id, user_id)
       end
 
+      # unsubscribe to the room chat
+      PubSub.unsubscribe("chat:" <> current_room_id)
+
```
Comment (ityonemo) [inline kousa/lib/kousa/room.ex:531]: you unsubscribe when you leave.

File: kousa/lib/onion/chat.ex [hunk 4]
```diff
@@ -177,6 +147,62 @@ defmodule Onion.RoomChat do
...
+  defp should_throttle(_, _), do: false
+
+  defp dispatch_message(payload, state) do
+    case payload.whisperedTo do
+      [] ->
+        PubSub.broadcast("chat:" <> state.room_id, %Broth.Message{
+          operator: "chat:send",
+          payload: payload
+        })
+
```
Comment (dark-elixir-shibe) [inline kousa/lib/onion/chat.ex:186]: this is where pubsub messages are broadcast out to all of the listening websocket processes.

**Pull request merged**